Add function to detect floating windows in ManageHook

andrea-berling opened the pull request:
This PR adds a function to `ManageHooks` to check whether the window to be managed is going to be a floating window or not. This feature and the problem it addresses are described more in detail in issue #300. An example configuration and an example use case are described in the issue. Under suggestion of @slotThe I decided to turn that issue into an actual PR for xmonad core. I've run the added code against the unit tests and nothing broke. I also made sure to test it manually using Xephyr to confirm that it was in fact doing what it tries to achieve.

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [x] I've confirmed these changes don't belong in xmonad-contrib instead

  - [x] I've considered how to best test these changes (property, unit,
        manually, ...) and concluded: unit testing + manual testing

  - [x] I updated the `CHANGES.md` file

andrea-berling committed:
Add function to detect floating windows in ManageHook

Add a willFloat function to deteect whether the managed window will be
floating or not

Add description of added change to CHANGES.md

geekosaur commented on src/XMonad/ManageHook.hs:
```diff
@@ -106,6 +106,14 @@ getStringProperty d w p = do
 
+-- | Return whether the window will be a floating window or not
+willFloat :: Query Bool
+willFloat = ask >>= \w -> liftX $ withDisplay $ \d -> do
```
Would it make more sense to refactor the code in XMonad.Operations to separate out its check, and then use that here?

andrea-berling replied:
It totally does, I did just that in my last commit. Thanks for pointing it out, I completely missed it the first time around :slightly_smiling_face:

geekosaur reviewed (approved):
Aside from my question above (which is not critical), this looks okay to me.

andrea-berling committed:
Factor our common logic for floating windows

Factour out the code used to detect whether a window should be floating
in Operations.hs in a new function named isFixedSizeOrTransient

Modify willFloat to use the factored out code from Opeartions.hs

geekosaur reviewed (approved):
LGTM!

slotThe reviewed (approved):
LGTM as well!  I will merge in a few if no one else objects

slotThe commented:
Thank you!
